cli/azd/internal/scaffold/funcs.go

```diff
@@ -81,6 +83,16 @@ func AlphaSnakeUpper(name string) string {
 	return sb.String()
 }
 
+func isAllUpperCase(c string) bool {
```

Comment:
If you are keeping this function (I'd recommend not to), add unit tests for it specifically and not thru `containerAppName()`.

Reply:
I think we are okay here.

`isAllUpperCase` shares the same algorithmic complexity (negation) of `isAnyLowerCase`

and `isAnyLowerCase` is defined as `'a' <= c <= 'z'`, which is trivial...